tools/flake8-torch/flake8_torch/checker.py

```diff
@@ -42,27 +102,110 @@ class TorchChecker:
 
     # The parameters need to have these exact names.
     # See https://flake8.pycqa.org/en/latest/plugin-development/plugin-parameters.html
-    # tree is unused but the plugin doesn't work without it.
+    # `tree` is unused, but the plugin doesn't work without it.
     def __init__(self, tree, lines):
-        self.deprecated_config = {}
-        with open(Path(__file__).absolute().parent / "deprecated_symbols.yaml") as f:
-            for item in yaml.load(f, yaml.SafeLoader):
-                self.deprecated_config[item["name"]] = item
-
         module = cst.parse_module("".join(lines))
-        visitor = TorchVisitor()
-        self.module = cst.MetadataWrapper(module)
-        self.module.visit(visitor)
-        self.func_calls = visitor.call_info
-
-    def _check_deprecated_functions(self):
-        for (name, lineno, col_offset) in self.func_calls:
-            if name in self.deprecated_config:
-                if self.deprecated_config[name]["remove_pr"] is None:
-                    message = f"TOR101 Use of deprecated function {name}"
-                else:
-                    message = f"TOR201 Use of removed function {name}"
-                yield ((lineno, 1 + col_offset, message, TorchChecker))
+        visitor = TorchVisitor(_read_deprecated_config())
+        module = cst.MetadataWrapper(module, unsafe_skip_copy=True)
+        module.visit(visitor)
+        self.violations = visitor.violations
 
     def run(self):
-        yield from self._check_deprecated_functions()
+        for violation in self.violations:
+            yield violation.flake8_result()
+
+
+class TorchCodemod(codemod.Codemod):
+    def transform_module_impl(self, module: cst.Module) -> cst.Module:
+        visitor = TorchVisitor(_read_deprecated_config())
+        fixes_count = 0
+
+        # We use `unsafe_skip_copy`` here not only to save some time, but
+        # because `deep_replace`` is identity-based and will not work on
+        # the original module if the wrapper does a deep copy:
+        # in that case we would need to use `wrapped_module.module`
+        # instead of `module`.
+        wrapped_module = cst.MetadataWrapper(module, unsafe_skip_copy=True)
+        wrapped_module.visit(visitor)
+        new_module = module
+
+        violations = visitor.violations
+        for violation in violations:
+            if violation.replacement is not None:
+                new_module = new_module.deep_replace(
+                    violation.node, violation.replacement
+                )
+                fixes_count += 1
+            try:
+                path = Path(self.context.filename).relative_to(Path.cwd())
+            except ValueError:
+                # Not a subpath of a current dir, use absolute path
+                path = self.context.filename
+            print(f"{path}{violation.codemod_result()}")
```

Comment:
I guess it's a debug print?
```suggestion
```

Reply:
No, it's the output of the checker when it runs in standalone mode (not as a flake8 plugin).